### PR TITLE
avoid duplicate dictionary lookups

### PR DESCRIPTION
**Pull request by SimonCropp**
minor perf improvement, and IMO improves readability

### CODE REVIEW

File: src/NUnitFramework/framework/Api/FrameworkController.cs [hunk 1]
```diff
@@ -114,17 +114,19 @@ private void Initialize(string assemblyNameOrPath, IDictionary settings)
...
+                    InternalTrace.Initialize((TextWriter)textWriterValue, traceLevel);
+                }
                 else
                 {
-                    var workDirectory = Settings.ContainsKey(FrameworkPackageSettings.WorkDirectory)
-                        ? (string)Settings[FrameworkPackageSettings.WorkDirectory]
-                        : Directory.GetCurrentDirectory();
+                    var workDirectory = Settings.TryGetValue(FrameworkPackageSettings.WorkDirectory, out var workDirectoryValue) ?
+                        (string)workDirectoryValue :
+                        Directory.GetCurrentDirectory();
```
Comment (jnm2) [inline src/NUnitFramework/framework/Api/FrameworkController.cs:129]: It's harder for me to scan code when the operators aren't at the start of the line. Was this part of the change intentional?

**Review by jnm2 — APPROVED** [review]

File: src/NUnitFramework/framework/Api/FrameworkController.cs (hunk 1 above)
Comment (SimonCropp) [reply src/NUnitFramework/framework/Api/FrameworkController.cs:129]: nope. will move to start

Comment (jnm2) [reply src/NUnitFramework/framework/Api/FrameworkController.cs]: Can you move the `:` back too?

File: src/NUnitFramework/framework/Api/FrameworkController.cs [hunk 2]
```diff
@@ -114,17 +114,19 @@ private void Initialize(string assemblyNameOrPath, IDictionary settings)
...
+                    InternalTrace.Initialize((TextWriter)textWriterValue, traceLevel);
+                }
                 else
                 {
-                    var workDirectory = Settings.ContainsKey(FrameworkPackageSettings.WorkDirectory)
-                        ? (string)Settings[FrameworkPackageSettings.WorkDirectory]
-                        : Directory.GetCurrentDirectory();
+                    var workDirectory = Settings.TryGetValue(FrameworkPackageSettings.WorkDirectory, out var workDirectoryValue)
+                        ? (string)workDirectoryValue :
+                        Directory.GetCurrentDirectory();
```
Comment (mikkelbu) [inline src/NUnitFramework/framework/Api/FrameworkController.cs:129]:
```suggestion
                        ? (string)workDirectoryValue
                        : Directory.GetCurrentDirectory();
```

File: src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs [hunk 3]
```diff
@@ -310,14 +315,14 @@ private void CreateTestExecutionContext(ITestListener listener)
...
-            if (Settings.ContainsKey(FrameworkPackageSettings.DefaultUICulture))
-                Context.CurrentUICulture = new CultureInfo((string)Settings[FrameworkPackageSettings.DefaultUICulture], false);
-            if (Settings.ContainsKey(FrameworkPackageSettings.StopOnError))
-                Context.StopOnError = (bool)Settings[FrameworkPackageSettings.StopOnError];
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultTimeout, out var timeout))
+                Context.TestCaseTimeout = (int)timeout;
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultCulture, out var culture))
+                Context.CurrentCulture = new((string)culture, false);
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultUICulture, out var uiCulture))
+                Context.CurrentUICulture = new((string)uiCulture, false);
```
Comment (mikkelbu) [inline src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs:323]: I don't think we use these inferred constructors anywhere else (at least at the moement), and I find the code harder to read when there is no type on either side of the assignment, so I think we should keep the old constructors (at least for now).
```suggestion
                Context.CurrentCulture = new CultureInfo((string)culture, false);
            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultUICulture, out var uiCulture))
                Context.CurrentUICulture = new CultureInfo((string)uiCulture, false);
```

**Review by mikkelbu — APPROVED** [review]
Looks good to me. I only have a couple of minor comments.

**Review by stevenaw — APPROVED** [review]

**Comment by stevenaw** [comment]
LGTM too. Thanks @SimonCropp 👍

**Pull request merged**